### ISSUE TITLE
[Bug] pre_uninstall should be run before the "application is still running" checks

### ISSUE THREAD

**Issue by beerpiss** [ScoopInstaller/Scoop#4961]
#### Current Behavior

Currently, `pre_uninstall` is run **after** `test_running_process`, which blocks the uninstallation if the app spawns any services.

#### Expected Behavior

I'd expect `pre_uninstall` to manage teardown work such as stopping services so that the uninstallation isn't blocked by running processes.

#### Additional context/output

An example of uninstalling [`itunes-np`](https://github.com/beerpiss/scoop-bucket/blob/main/experimental/itunes-np.json):
```
❯ sudo scoop uninstall -g itunes-np                  
Uninstalling 'itunes-np' (12.12.4.1).
WARN  Application "itunes-np" is still running. Scoop is configured to ignore this condition.
Running pre_uninstall script...
Running uninstaller script...
Removing shortcut C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Scoop Apps\iTunes.lnk
Unlinking C:\ProgramData\scoop\apps\itunes64-np\current
'itunes-np' was uninstalled.
```
As shown, `pre_uninstall` is run after `test_running_processes`, and so uninstallation would've been blocked if `scoop config ignore_running_processes` were `False`.

#### Possible Solution

Move `pre_uninstall` to before `test_running_processes`, to the very top, before any uninstalling prep work is run. This way, it also matches `post_uninstall`, which is run at the very end of the uninstall process.

### System details

**Windows version:** 10.0.19041.1499

**OS architecture:** 64bit

**PowerShell version:** 7.2.2

#### Scoop Configuration
```json
{
  "lastUpdate": "2022-06-02T12:19:59.4769844+07:00",
  "SCOOP_REPO": "https://github.com/ScoopInstaller/Scoop/",
  "SCOOP_BRANCH": "develop",
  "aria2-enabled": true,
  "MSIEXTRACT_USE_LESSMSI": true,
  "SCOOP_DEBUG": true,
  "cachePath": "D:\\Temp\\scoop\\cache",
  "aria2-warning-enabled": false,
  "aria2-split": 16,
  "aria2-max-connection-per-server": 16,
  "aria2-min-split-size": "1M",
  "ignore_running_processes": true,
  "alias": {
    "upgrade": "scoop-upgrade"
  }
}
```

**Comment by chawyehsu** [comment]
Good catch, could you create a PR to fix it?

**Comment by beerpiss** [comment]
> Good catch, could you create a PR to fix it?

Created #4962

**Comment by beerpiss** [comment]
Fixed by #4962.
